.github/workflows/test.yml

```diff
@@ -12,10 +12,31 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+
       - uses: actions/setup-go@v2
         with:
           go-version: 1.17
+
       - name: Start minikube
         uses: medyagh/setup-minikube@master
+
+      - name: Set cache paths
+        id: go-cache-paths
+        run: |
+          echo "::set-output name=go-build::$(go env GOCACHE)"
+          echo "::set-output name=go-mod::$(go env GOMODCACHE)"
+
+      - name: Go build cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.go-cache-paths.outputs.go-build }}
+          key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}
+
+      - name: Go mod cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.go-cache-paths.outputs.go-mod }}
+          key: ${{ runner.os }}-go-mod-${{ hashFiles('**/go.sum') }}          
```

Comment:
Merge these two, `path` can be a newline delimited list of cache paths

Reply:
Include `restore-keys` with `${{ runner.os }}-go-` to add partial caching. When only specifying `key`, you either get a perfect hit (esp. for mod) or none. But if you've actually changed a single dependency in `go.sum` or updated a single package, you'd still benefit from downloading _any_ recent cache, which will still be valid for other modules/packages.

Reply:
See https://github.com/actions/cache/blob/main/examples.md#go---modules which implements Go caching correctly, minus the use of `go env` to get the OS-specific path.